src/yarpWholeBodySensors.cpp

```diff
@@ -474,29 +588,41 @@ bool yarpWholeBodySensors::loadAccelerometerInfoFromConfig(const Searchable& opt
             return false;
         }
     }
-
     infos.resize(list.size());
 
-    for(int acc_index = 0; acc_index < (int)list.size(); acc_index++ )
+//     std::cout<<"-----------------------\nRecovering IDList IDs and lookinf for rest of info in list of size "<<list.size()<<"\n";
+    for(int acc_index =0; acc_index < (int)list.size(); acc_index++ )
     {
         wbi::ID acc_ID;
         list.indexToID(acc_index,acc_ID);
-        yarp::os::Bottle * port = info_lists.find(acc_ID.toString()).asList();
-        //std::cout << port->toString() << std::endl;
-        if( port == NULL
-            || port->size() != 2
-            || !(port->get(0).isString())
-            || !(port->get(1).isString()) )
+//         std::cout <<"---------------- AccID : "<< acc_ID.toString() << std::endl;
+        yarp::os::Bottle * sensorInfo = info_lists.find(acc_ID.toString()).asList();
+//         std::cout <<"---------------- Acc contents"<< sensorInfo->toString() << std::endl;
+        if( sensorInfo == NULL
+            || sensorInfo->size() != 3
+            || !(sensorInfo->get(0).isString())
+            || !(sensorInfo->get(1).isInt())
+            || !(sensorInfo->get(2).isString()) )
         {
             std::cout << "yarpWbi::loadAccelerometerInfoFromConfig error: " << info_lists.toString() << " has a malformed element" << std::endl;
             return false;
         }
-        std::string accelerometer_type = port->get(0).asString();
-        std::string accelerometer_type_option = port->get(1).asString();
+        infos[acc_index].port_name = std::string("/icub/" ) + 
+                                                sensorInfo->get(0).asString() +
+                                                std::string("/inertialMTB");
```

Comment:
Can you keep the complete port name in the configuration file (or at least the part after `/icub` ? In this way we can use this code for both the ETH and CAN robots.